semihost/open.c

```diff
@@ -69,8 +69,10 @@ open(const char *pathname, int flags, ...)
 	default:
 		if (flags & O_TRUNC)
 			semiflags = SH_OPEN_W_PLUS_B;	/* 'wb+' */
-		else
+		else if (flags & O_APPEND)
```

Comment:
Hrm. None of the modes which create the file are checking for O_CREAT. That seems wrong.